src/constructors.jl

```diff
@@ -66,174 +66,38 @@ function Base.Array(chn::Chains,
-    sections = sections isa AbstractArray ? sections : [sections]
-    sections = showall ? [] : sections
-    section_list = length(sections) == 0 ?
-        sort_sections(chn) :
-        sections
+    sections = showall ? keys(chn.name_map) : sections
+    chn = Chains(chn, sections)
 
-    # If we actually have missing values, we can't remove
-    # Union{Missing}.
     remove_missing_union = remove_missing_union ?
         all(x -> !ismissing(x), chn.value) :
         remove_missing_union
 
-    d, p, c = size(chn.value.data)
-
-    local b
-    if append_chains
-        first_parameter = true
-        for section in section_list
-            for par in chn.name_map[section]
-                x = get(chn, Symbol(par))
-                d, c = size(x[Symbol(par)])
-                if first_parameter
-                    if remove_missing_union
-                        b = reshape(convert(Array{Float64}, x[Symbol(par)]), d*c)[:, 1]
-                    else
-                        b = reshape(x[Symbol(par)], d*c)[:, 1]
-                    end
-                    p == 1 && (b = reshape(b, size(b, 1)))
-                    first_parameter = false
-                else
-                    if remove_missing_union
-                        b = hcat(b, reshape(convert(Array{Float64}, x[Symbol(par)]), d*c)[:, 1])
-                    else
-                        b = hcat(b, reshape(x[Symbol(par)], d*c)[:, 1])
-                    end
-                end
-            end
+    arr = if append_chains
+        if remove_missing_union
+            vcat([convert(Array{Real}, chn.value.data[:,:,i]) for i in 1:size(chn, 3)]...)
+        else
+            vcat([chn.value.data[:,:,i] for i in 1:size(chn, 3)]...)
         end
     else
-        b=Vector(undef, c)
-        for i in 1:c
-            first_parameter = true
-            for section in section_list
-                for par in chn.name_map[section]
-                    x = get(chn, Symbol(par))
-                    d, c = size(x[Symbol(par)])
-                    if first_parameter
-                        if remove_missing_union
-                            b[i] = convert(Array{Real}, x[Symbol(par)][:, i])
-                        else
-                            b[i] = x[Symbol(par)][:, i]
-                        end
-                        p == 1 && (b[i] = reshape(b[i], size(b[i], 1)))
-                        first_parameter = false
-                    else
-                        if remove_missing_union
-                            b[i] = hcat(b[i], convert(Array{Real}, x[Symbol(par)][:, i]))
-                        else
-                            b[i] = hcat(b[i], x[Symbol(par)][:, i])
-                        end
-                    end
-                end
-            end
+        if remove_missing_union
+            [convert(Array{Real}, chn.value.data[:,:,i]) for i in 1:size(chn, 3)]
+        else
+            [chn.value.data[:,:,i] for i in 1:size(chn, 3)]
```

Comment:
I'm wondering if one should remove the keyword argument `append_chains` and instead use two different functions since the different return types make the function type unstable.

Reply:
What would be a good signature? `jagged_array`? I'm also not a fan of calling this function `Array`, since the actual return value for this should be `chain.value.data`, which is already a 3D array.

Maybe this should be `stacked_array` and `jagged_array`, but I think there's better names we can come up with?

Reply:
Let's leave this for now and come back to it later.